packages/experiment-tag/src/experiment.ts

```diff
@@ -61,13 +64,25 @@ export const initializeExperiment = (apiKey: string, initialFlags: string) => {
       const parsedFlags = JSON.parse(initialFlags);
       parsedFlags.forEach((flag: EvaluationFlag) => {
         if (flag.key in urlParams && urlParams[flag.key] in flag.variants) {
-          flag.segments = [
-            {
-              metadata: { segmentName: 'preview' },
-              variant: urlParams[flag.key],
-            },
-          ];
+          // Keep page targeting segments
+          const pageTargetingSegments = flag.segments.filter((segment) =>
+            isPageTargetingSegment(segment),
+          );
+
+          // Create or update the preview segment
+          const previewSegment = {
+            metadata: { segmentName: 'preview' },
+            variant: urlParams[flag.key],
+          };
+
+          flag.segments = [...pageTargetingSegments, previewSegment];
```

Comment:
I have a feeling our initial implementation is too complicated. Maybe we just force apply the specific flag & variant actions.

Reply:
For another time though. Just a thought